qcodes/dataset/data_set.py

```diff
@@ -322,6 +324,31 @@ def __init__(self, path_to_db: Optional[str] = None,
                 active_datasets=set())
             _WRITERS[self.path_to_db] = ws
 
+    def prepare(
+        self,
+        station: "Optional[Station]",
+        interdeps: InterDependencies_,
+        write_in_background: bool,
+        shapes: Shapes = None,
```

Comment:
implicit optional? :)
```suggestion
        shapes: Optional[Shapes] = None,
```

Reply:
Shapes is already `Optional[Dict[str, Tuple[int, ...]]]`